File: WondayWall.iOS/WondayWall/Models/NewsTopicItem.swift
import Foundation

// ニューストピック
struct NewsTopicItem: Codable, Identifiable {
    var id: String
    let title: String
    let summary: String?
    let url: String?
    let publishedAt: Date
    let ogpImageUrl: String?
    let sourceRssUrl: String?

    init(
        id: String,
        title: String,
        summary: String? = nil,
        url: String? = nil,
        publishedAt: Date,
        ogpImageUrl: String? = nil,
        sourceRssUrl: String? = nil
    ) {
        self.id = id
        self.title = title
        self.summary = summary
        self.url = url
        self.publishedAt = publishedAt
        self.ogpImageUrl = ogpImageUrl
        self.sourceRssUrl = sourceRssUrl
    }
}

File: WondayWall.Android/app/src/main/kotlin/com/studiofreesia/wondaywall/models/NewsTopicItem.kt
package com.studiofreesia.wondaywall.models

import kotlinx.datetime.Instant
import kotlinx.serialization.Serializable

// ニューストピック（RSS/フィードから取得した記事）
@Serializable
data class NewsTopicItem(
    // 一意ID（RSS の guid またはURL から生成）
    val id: String,
    // 記事タイトル
    val title: String,
    // 記事の概要
    val summary: String?,
    // 記事URL
    val url: String?,
    // 公開日時
    val publishedAt: Instant,
    // OGP画像URL（取得できた場合）
    val ogpImageUrl: String?,
)
